src/environment/conda_prefix.rs

```diff
@@ -36,96 +38,152 @@ pub struct CondaPrefixUpdated {
     pub python_status: Box<PythonStatus>,
 }
 
-#[derive(Clone)]
 /// A task that updates the prefix for a given environment.
-pub struct CondaPrefixUpdater<'a> {
-    pub group: GroupedEnvironment<'a>,
+pub struct CondaPrefixUpdaterInner {
+    pub channels: Vec<ChannelUrl>,
+    pub name: GroupedEnvironmentName,
+    pub client: ClientWithMiddleware,
+    pub prefix: Prefix,
+    pub virtual_packages: Vec<GenericVirtualPackage>,
     pub platform: Platform,
     pub package_cache: PackageCache,
     pub io_concurrency_limit: IoConcurrencyLimit,
     pub build_context: BuildContext,
+
+    /// A flag that indicates if the prefix was created.
+    created: AsyncOnceCell<CondaPrefixUpdated>,
 }
 
-impl<'a> CondaPrefixUpdater<'a> {
+impl CondaPrefixUpdaterInner {
     /// Creates a new prefix task.
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
-        group: GroupedEnvironment<'a>,
+        channels: Vec<ChannelUrl>,
+        name: GroupedEnvironmentName,
+        client: ClientWithMiddleware,
+        prefix: Prefix,
+        virtual_packages: Vec<GenericVirtualPackage>,
```

Comment:
Final thing, maybe because we are re-using this a lot, you could add a `Builder` which is a seperate struct in which a `GroupedEnviromnent` needs to be passed. This way we can re-use the extraction code, and if you keep the `new` method public you could keep using this in the tests if you want :)